scrabble.js

```diff
@@ -1,14 +1,138 @@
+const score_chart = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10
+};
 const Scrabble = {
   score: function(word) {
-    // TODO: implement score
-  }
+    format = /^[a-zA-Z]+$/;
+    word = word.toUpperCase();
+    if (format.test(word) && word.length <= 7 && word != '') {
+      let letter, i, sum = 0;
+      for (i = 0; i < word.length; i++) {
+        letter = word[i];
+        sum += score_chart[letter];
+      } // for
+      if (word.length === 7) {
+        return parseInt(sum + 50);
+      } else {
+        return parseInt(sum);
+      }
+    } // if
+    else {
+      throw new Error(`Invalid word to score "${ word }"`);
+    } // else
+  }, // score
 
-  // TODO: add the highestScoreFrom method
+  highestScoreFrom: function(arrayOfWords) {
+    if (arrayOfWords.length === 0) {
+      throw new Error('No words to score');
+    } else if (arrayOfWords.length === 1){
+      return arrayOfWords[0];
+    }
+    let tie = [];
+    let max_so_far = -1;
+    for(let word of arrayOfWords) {
```

Comment:
This algorithm ends up being a little over-complicated. By my reading, your steps are:

1. Calculate a score for every word
1. Find those words that share the highest score
1. Use apply tie-breaking logic to those words

A simpler algorithm is to iterate through the list of words, keeping track of the best word / score seen so far. For each word:
- If the current word has a higher score than the best word, it's the new best word
- If the best word has a higher score, keep it
- If the two are tied, apply tie-breaking logic, favoring the existing best word.

This single-pass approach is more concise and (in my opinion) easier to understand.